### PR TITLE
bpo-46586: Fix more erroneous doc links to builtins

### PR DESCRIPTION
**Pull request by meersuri**
https://bugs.python.org/issue46586

### CODE REVIEW

**Comment by AlexWaygood** [comment]
Hi, it looks like you need to merge `origin/main` into your branch here — the `enum.property` fix is still showing up in this PR, even though that one's already been committed to `main` 🙂

File: Doc/library/io.rst [hunk 1]
```diff
@@ -55,7 +55,7 @@ the backing store is natively made of bytes (such as in the case of a file),
 encoding and decoding of data is made transparently as well as optional
 translation of platform-specific newline characters.
 
-The easiest way to create a text stream is with :meth:`open()`, optionally
+The easiest way to create a text stream is with :meth:`~io.open`, optionally
```
Comment (merwok) [inline Doc/library/io.rst]: I am not sure this one is a problem; built-in open is io.open after all!  We certainly don’t recommend that people must import io and use io.open.

File: Doc/library/zipfile.rst [hunk 2]
```diff
@@ -289,7 +289,7 @@ ZipFile Objects
       compressed text files in :term:`universal newlines` mode.
 
    .. versionchanged:: 3.6
-      :meth:`open` can now be used to write files into the archive with the
+      :meth:`~ZipFile.open` can now be used to write files into the archive with the
```
Comment (merwok) [inline Doc/library/zipfile.rst]: Not sure that I would hide `ZipFile` here; being explicit would benefit readers.

File: Doc/library/io.rst (hunk 1 above)
Comment (JelleZijlstra) [reply Doc/library/io.rst]:
```suggestion
The easiest way to create a text stream is with :meth:`open()`, optionally
```

File: Doc/library/zipfile.rst (hunk 2 above)
Comment (JelleZijlstra) [reply Doc/library/zipfile.rst]:
```suggestion
      :meth:`ZipFile.open` can now be used to write files into the archive with the
```

File: Doc/library/io.rst [hunk 3]
```diff
@@ -77,7 +77,7 @@ objects.  No encoding, decoding, or newline translation is performed.  This
 category of streams can be used for all kinds of non-text data, and also when
 manual control over the handling of text data is desired.
 
-The easiest way to create a binary stream is with :meth:`open()` with ``'b'`` in
+The easiest way to create a binary stream is with :meth:`~io.open` with ``'b'`` in
```
Comment (JelleZijlstra) [inline Doc/library/io.rst]:
```suggestion
The easiest way to create a binary stream is with :meth:`open()` with ``'b'`` in
```

**Review by JelleZijlstra — APPROVED** [review]

File: Doc/library/io.rst (hunk 3 above)
Comment (AlexWaygood) [reply Doc/library/io.rst]: I think the :meth: role automatically adds the parentheses afterwards 🙂

File: Doc/library/fileinput.rst [hunk 4]
```diff
@@ -224,5 +224,5 @@ The two following opening hooks are provided by this module:
       Added the optional *errors* parameter.
 
    .. deprecated:: 3.10
-      This function is deprecated since :func:`input` and :class:`FileInput`
+      This function is deprecated since :func:`~fileinput.input` and :class:`FileInput`
```
Comment (merwok) [inline Doc/library/fileinput.rst]: I would not hide the module name here.
(I do not mind if we have `fileinput.input` and `FileInput` without module, the second one is not ambiguous.)

File: Doc/library/urllib.request.rst [hunk 5]
```diff
@@ -737,7 +737,7 @@ The following attribute and methods should only be used by classes derived from
 
    This method, if implemented, will be called by the parent
    :class:`OpenerDirector`.  It should return a file-like object as described in
-   the return value of the :meth:`open` of :class:`OpenerDirector`, or ``None``.
+   the return value of the :meth:`~OpenerDirector.open` of :class:`OpenerDirector`, or ``None``.
```
Comment (merwok) [inline Doc/library/urllib.request.rst]:
```suggestion
   the return value of the method :meth:`~OpenerDirector.open` of :class:`OpenerDirector`, or ``None``.
```

Comment (AlexWaygood) [reply Doc/library/urllib.request.rst]: Counter-suggestion ;)

```suggestion
   the return value of the :meth:`~OpenerDirector.open` method of :class:`OpenerDirector`, or ``None``.
```

File: Doc/library/fileinput.rst (hunk 4 above)
Comment (JelleZijlstra) [reply Doc/library/fileinput.rst]:
```suggestion
      This function is deprecated since :func:`fileinput.input` and :class:`FileInput`
```

**Comment by JelleZijlstra** [comment]
@merwok let's keep that for another PR, so we can land this one.

**Comment by merwok** [comment]
I don’t really get why split these fixes into three changes here, two changes there (and the original ticket about `enum` has already had its scope widered), but 🤷🏽 it’s up to you!  Better to merge some fixes than none 🙂

**Pull request merged**